marlowe/test/Spec/Marlowe/Serialization/ExtendedJson.hs

```diff
@@ -0,0 +1,50 @@
+-----------------------------------------------------------------------------
+--
+-- Module      :  $Headers
+-- License     :  Apache 2.0
+--
+-- Stability   :  Experimental
+-- Portability :  Portable
+--
+-- | This suite tests the Json serialization of the Marlowe extended module
+--
+-----------------------------------------------------------------------------
+{-# LANGUAGE OverloadedStrings #-}
+
+module Spec.Marlowe.Serialization.ExtendedJson
+  ( tests
+  ) where
+
+import Data.Aeson (eitherDecodeFileStrict)
+import Language.Marlowe.Extended.V1 (Contract, Module)
+import System.Directory (getCurrentDirectory)
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.HUnit (assertFailure, testCase)
+
+tests :: TestTree
+tests = testGroup "Extended Contract Serialization"
+    [ testCase "Golden Swap contract" testGoldenSwapContract
+    , testCase "Golden Swap module" testGoldenSwapModule
+    ]
+
+-- TODO: Do a small non-property round-trip with example contracts.
+
+
+-- | Checks that we can decode the Golden JSON Contract for Swap
+testGoldenSwapContract :: IO ()
+testGoldenSwapContract = do
+    filePath <- (\d -> d ++ "/test/Spec/Marlowe/Serialization/golden/swap-contract.json") <$> getCurrentDirectory
```

Comment:
You don't really need to prepend the current directory and `/`, since `eitherDecodeFileStrict` will read a path relative the current directory anyway. Also the `filepath` package provides OS-independent path navigation.

Reply:
done